exporters/metrics/src/main/java/com/google/cloud/opentelemetry/metric/MetricExporter.java

```diff
@@ -96,62 +93,73 @@ private static MetricExporter createWithCredentials(
+          logger.error(
+          "Metric type {} not supported. Only gauge and cumulative types are supported.",
+          metricData.getType());
+          continue;
       }
-      MetricData.Point metricPoint = metricData.getPoints().iterator().next();
 
-      MetricDescriptor descriptor = mapMetricDescriptor(metricData, metricPoint);
-      if (descriptor == null) {
-        continue;
-      }
-      metricServiceClient.createMetricDescriptor(
-          CreateMetricDescriptorRequest.newBuilder()
-              .setName(PROJECT_NAME_PREFIX + projectId)
-              .setMetricDescriptor(descriptor)
-              .build());
-
-      MetricWithLabels updateKey =
-          new MetricWithLabels(descriptor.getType(), metricPoint.getLabels());
-
-      // Cloud Monitoring API allows, for any combination of labels and
-      // metric name, one update per WRITE_INTERVAL seconds
-      long pointCollectionTime = metricPoint.getEpochNanos();
-      if (lastUpdatedTime.containsKey(updateKey)
-          && pointCollectionTime
-              <= lastUpdatedTime.get(updateKey) / NANO_PER_SECOND + WRITE_INTERVAL_SECOND) {
-        continue;
+      for (final MetricDescriptor descriptor: builder.getDescriptors()) {
+        // TODO: limit this ONCE per JVM.
+        metricServiceClient.createMetricDescriptor(
+            CreateMetricDescriptorRequest.newBuilder()
+                .setName(PROJECT_NAME_PREFIX + projectId)
+                .setMetricDescriptor(descriptor)
+                .build());
       }
 
-      Metric metric = mapMetric(metricPoint, descriptor.getType());
-      Point point = mapPoint(metricData, metricPoint, updateKey, lastUpdatedTime);
-      if (point == null) {
-        continue;
-      }
-
-      allTimesSeries.add(
-          TimeSeries.newBuilder()
-              .setMetric(metric)
-              .addPoints(point)
-              .setResource(mapResource(projectId))
-              .setMetricKind(descriptor.getMetricKind())
-              .build());
-    }
-    createTimeSeriesBatch(metricServiceClient, ProjectName.of(projectId), allTimesSeries);
-    if (allTimesSeries.size() < metrics.size()) {
+      // TODO: Filter metrics by last updated time....
+      // MetricWithLabels updateKey =
+      //     new MetricWithLabels(descriptor.getType(), metricPoint.getLabels());
+
+      // // Cloud Monitoring API allows, for any combination of labels and
+      // // metric name, one update per WRITE_INTERVAL seconds
+      // long pointCollectionTime = metricPoint.getEpochNanos();
+      // if (lastUpdatedTime.containsKey(updateKey)
+      //     && pointCollectionTime
+      //         <= lastUpdatedTime.get(updateKey) / NANO_PER_SECOND + WRITE_INTERVAL_SECOND) {
+      //   continue;
+      // }
+    }    
+    List<TimeSeries> series = builder.getTimeSeries();
+    createTimeSeriesBatch(metricServiceClient, ProjectName.of(projectId), series);
+    // TODO: better error reporting.
+    if (series.size() < metrics.size()) {
```

Comment:
Re above comment about each `MetricData` having a list of points; if that's not the case, this would report failure. Should we just remove the check for now if its not adding anything?

Reply:
I'm trying to not change the behavior right before 1.0.  I agree this needs to get fixed, but I'd rather spend some time on it.

Given metrics is alpha, I think the TODO and a bug to fix is the right way to go for now.